src/ipelib/dynamo/module_weimer2005Ipe.f

```diff
@@ -0,0 +1,806 @@
+         pos(i) = tmat(1,i)*a + tmat(2,i)*b + tmat(3,i)*stc
+      enddo
+  
+      latout = asin(pos(3))*rad2deg
+      lonout = atan2(pos(2),pos(1))*rad2deg
+
+!   write(6,"('dorotation: latin,lonin=',2f9.4,' latout,lonout=',2f9.4)") &
+!     latin,lonin,latout,lonout
+  
+      end subroutine dorotation2005Ipe
+!-----------------------------------------------------------------------
+      subroutine interpol_quad2005Ipe(v,x,u,p)
+!
+! f90 translation of IDL function interpol(v,x,u,/quadratic)
+!
+      implicit none
+!
+! Args:
+      real*8,intent(in) :: v(:),x(:),u(:)
+      real*8,intent(out) :: p(:)
+!
+! Local:
+      integer :: nv,nx,nu,i,ix
+      real*8 :: x0,x1,x2
+!
+      nv = size(v)
+      nx = size(x)
+      nu = size(u)
+      if (nx /= nv) then
+         write(6,"('>>> interpol_quad: nx /= nv: nx=',i4,' nv=',i4)")   &
+     &    nx,nv
+         p(:) = 0.
+         return
+      endif
+      do i=1,nu
+         ix = value_locate2005Ipe(x,u(i))
+         if (ix <= 1.or.ix >= nx) then
+!       write(6,"('>>> interpol_quad: ix out of range: nu=',i4,' ix=',i4)") nu,ix
+            p(i) = 0.
+            continue
+         endif
+         x1 = x(ix)
+         x0 = x(ix-1)
+         x2 = x(ix+1)
+         if(x0.eq.0..and.x1.eq.0.) then
+            p(i) = 0.0
+         else
+            p(i) = v(ix-1) * (u(i)-x1) * (u(i)-x2)/((x0-x1) * (x0-x2))+ &
+     &             v(ix)   * (u(i)-x0) * (u(i)-x2)/((x1-x0) * (x1-x2))+ &
+     &             v(ix+1) * (u(i)-x0) * (u(i)-x1)/((x2-x0) * (x2-x1))
+         endif
+      enddo
+!   write(6,"('interpol_quad: nu=',i4,' p=',/,(1pe12.4)") nu,p
+
+      end subroutine interpol_quad2005Ipe
+!-----------------------------------------------------------------------
+      integer function value_locate2005Ipe(vec,val)
+!
+! f90 translation of IDL function value_locate
+! Return index i into vec for which vec(i) <= val >= vec(i+1)
+! Input vec must be monotonically increasing
+!
+      implicit none
+!
+! Args:
+      real*8,intent(in) :: vec(:),val
+!
+! Local:
+      integer :: n,i
+!
+      value_locate2005Ipe = 0
+      n = size(vec)
+      if (val < vec(1)) return
+      if (val > vec(n)) then
+         value_locate2005Ipe = n
+         return
+      endif
+      do i=1,n-1
+         if (val >= vec(i) .and. val <= vec(i+1)) then
+            value_locate2005Ipe = i
+            return
+         endif
+      enddo
+
+      end function value_locate2005Ipe
+!-----------------------------------------------------------------------
+      real function lngamma2005Ipe(xx)
+!
+! This is an f90 translation from C code copied from 
+! www.fizyka.umk.pl/nrbook/c6-1.pdf (numerical recipes gammln)
+!
+      implicit none
+      real*8,intent(in) :: xx
+      real*8 :: x,y,tmp,ser
+      real*8 :: cof(6) = (/76.18009172947146, -86.50532032941677        &
+     &,24.01409824083091                                                &
+     &,-1.231739572450155, 0.1208650973866179e-2, -0.5395239384953e-5/)
```

Comment:
Define as parameter.

Reply:
Let's address the Weimer thing altogether when Zhuxiao merges in her Weimer2005_empirical branch. The same code is used in different parts of the model.